integration/tests/access/cohort2/observer_indexer_enabled_test.go

```diff
@@ -463,3 +664,396 @@ func (s *ObserverIndexerEnabledSuite) getRestEndpoints() []RestEndpointTest {
 		},
 	}
 }
+
+func (s *ObserverIndexerEnabledSuite) checkGetEventsForBlockIDsRPC(
+	ctx context.Context,
+	observerLocalResponse *accessproto.EventsResponse,
```

Comment:
looks like you pass the local response in a few checkers, and just the clients in others. why use different approaches?

if possible, it's cleaner to move all query logic into the checker so it's easy to see the requests all in one place.